src/components/DBEntryManagerPanel/AddOrEditDBEntry/index.tsx

```diff
@@ -0,0 +1,574 @@
+    const validations = entryFields.map(() => { return true; });
+    let validationError = '';
+
+    for (let i = 0; i < entryFields.length; i += 1) {
+      const field = entryFields[i];
+      const value = entry[field.objectKey];
+
+      if (field.required && !value) {
+        validations[i] = false;
+        validationError = `Please fill in the ${field.label} field`;
+        break;
+      }
+
+      if (field.objectKey === idPropName) {
+        if (entries.find((e) => { return e[idPropName] === value; })) {
+          validations[i] = false;
+          validationError = `An item with the ${field.label} ${value} already exists. ${field.label} must be unique.`;
+          break;
+        }
+      }
+
+      if (value) {
+        if (field.type === DBEntryFieldType.String) {
+          if (field.minNumChars && value.length < field.minNumChars) {
+            validations[i] = false;
+            validationError = `${field.label} must be at least ${field.minNumChars} characters long`;
+          } else if (field.maxNumChars && value.length > field.maxNumChars) {
+            validations[i] = false;
+            validationError = `${field.label} must be at most ${field.maxNumChars} characters long`;
+          }
+        } else if (field.type === DBEntryFieldType.Number) {
+          if (field.minNumber && value < field.minNumber) {
+            validations[i] = false;
+            validationError = `${field.label} must be at least ${field.minNumber}`;
+          } else if (field.maxNumber && value > field.maxNumber) {
+            validations[i] = false;
+            validationError = `${field.label} must be at most ${field.maxNumber}`;
+          }
+        } else if (field.type === DBEntryFieldType.StringArray || field.type === DBEntryFieldType.NumberArray) {
+          if (field.minNumElements && value.length < field.minNumElements) {
+            validations[i] = false;
+            validationError = `${field.label} must have at least ${field.minNumElements} values`;
+          } else if (field.maxNumElements && value.length > field.maxNumElements) {
+            validations[i] = false;
+            validationError = `${field.label} must have at most ${field.maxNumElements} values`;
+          } else if (field.type === DBEntryFieldType.NumberArray) {
+            for (let j = 0; j < value.length; j += 1) {
+              if (field.minNumber && value[j] < field.minNumber) {
+                validations[i] = false;
+                validationError = `${field.label} values must be at least ${field.minNumber}`;
+                break;
+              } else if (field.maxNumber && value[j] > field.maxNumber) {
+                validations[i] = false;
+                validationError = `${field.label} values must be at most ${field.maxNumber}`;
+                break;
+              }
+            }
+          }
+        }
+      }
+    }
+
+    const validationFailed = validations.includes(false);
```

Comment:
This is equivalent to `!!validationError`, since the empty string is falsey, so this variable is unnecessary.

Reply:
The `validations` array is also unneeded